ui/src/app/applications/components/application-parameters/application-parameters.tsx

```diff
@@ -299,7 +299,7 @@ export const ApplicationParameters = (props: {
             title: 'ENV',
             view: (
                 <div style={{marginTop: 15}}>
-                    {app.spec.source.plugin &&
+                    {app.spec.source?.plugin &&
                         (app.spec.source.plugin.env || []).map(val => (
```

Comment:
I think this can be shortened

```suggestion
                    {app.spec.source?.plugin?.env || []).map(val => (
```

Reply:
Indeed. Simplified.